### ISSUE TITLE
Support rendering slices to a StringBuilder

### ISSUE THREAD

**Issue by DamianEdwards** [DamianEdwards/RazorSlices#12]
E.g.

```csharp
public static ValueTask RenderAsync(this RazorSlice slice, StringBuilder stringBuilder, HtmlEncoder? htmlEncoder = null);
```

**Comment by DamianEdwards** [comment]
I think that's a fine approach. While it's going through a few layers (`IBufferWriter<byte>` -> `TextWriter` -> `StringWriter` -> `StringBuilder`) they should all be simply passing along to the next layer as much as possible.

You should be able to just return the task result of calling `RenderAsync` directly though, no need for the `IsCompletedSuccessfully` check as you don't need to go async.

**Comment by madeyoga** [comment]
Thank you for your response! I will proceed with the `StringWriter` approach. May I submit a pull request for this feature? Please let me know if you have any further suggestions or requirements for the implementation.

**Comment by DamianEdwards** [comment]
Please go ahead!

**Comment by DamianEdwards** [comment]
This was done and shipped in [v0.6.0](https://github.com/DamianEdwards/RazorSlices/releases/tag/v0.6.0)
